docs/literate/src/files/behind_the_scenes_simulation_setup.jl

```diff
@@ -0,0 +1,251 @@
+
+# In conclusion, the primary purpose of a `SemidiscretizationHyperbolic` is to collect equations,
+# the geometric representation of the domain, and approximation instructions, creating specialized
+# structures to interconnect these components in a manner that enables their utilization for
+# the numerical solution of partial differential equations (PDEs).
+
+# As evident from the earlier description of `SemidiscretizationHyperbolic`, it comprises numerous
+# functions called recursively. Without delving into details, the structure of the primary calls
+# are illustrated as follows:
+
+# ![SemidiscretizationHyperbolic_structure](https://github.com/trixi-framework/Trixi.jl/assets/119304909/8bf59422-0537-4d7a-9f13-d9b2253c19d7)
+
+# ## Overview of the [`semidiscretize`](@ref) function
+
+# At this stage, we have defined the equations and configured the domain's discretization. The
+# final step before solving is to select a suitable time span and apply the corresponding initial
+# conditions, which are already stored in the initialized `SemidiscretizationHyperbolic` object.
+
+# The purpose of the [`semidiscretize`](@ref) function is to wrap the semidiscretization as an
+# `ODEProblem` within the specified time interval. During this procedure the approximate solution
+#  is created at the given initial time via the specified `initial_condition` function from the 
+#  `SemidiscretizationHyperbolic` object. This `ODEProblem` can be subsequently passed to the
+# `solve` function from the [OrdinaryDiffEq.jl](https://github.com/SciML/OrdinaryDiffEq.jl) package
+# or to [`Trixi.solve`](@ref).
+
+ode = semidiscretize(semi, (0.0, 1.0));
+
+# The `semidiscretize` function involves a deep tree of subsequent calls, with the primary ones
+# explained below.
+
+
+# - `allocate_coefficients(mesh, equations, solver, cache)`
+
+#   To apply initial conditions, a data structure ("container") needs to be generated to store the
+#   initial values of the target variables for each node within each element.
+
+#   Since only one-dimensional `Array`s are `resize!`able in Julia, we use `Vector`s as an internal
+#   storage for the target variables and `resize!` them whenever needed, e.g. to change the number
+#   of elements. Then, during the solving process the same memory is reused by `unsafe_wrap`ping
+#   multi-dimensional `Array`s around the internal storage.
+
+# - `wrap_array(u_ode, semi)`
+
+#   As previously noted, `u_ode` is constructed as a 1D vector to ensure compatibility with
+#   OrdinaryDiffEq.jl. However, for internal use within Trixi.jl, identifying which part of the
+#   vector relates to specific variables, elements, or nodes can be challenging.
+
+#   This is why the `u_ode` vector is wrapped by the `wrap_array` function to create a
+#   multidimensional array `u`, with each dimension representing variables, nodes and elements.
+#   Consequently, navigation within this multidimensional array becomes noticeably easier.
```

Comment:
Should we be even more precise here and say how exactly the wrapped array looks like and which axis has which meaning (and size), cf. https://github.com/trixi-framework/Trixi.jl/blob/main/src/solvers/dg.jl#L651-L653?

Reply:
Do you think it is really necessary, I mean tutorial is about basic concepts used in Trixi.jl, should we dive in such technical aspects? I think it could be confusing....

Reply:
Hm I have no strong opinion - it does probably not hurt to have this here, but I also think that someone who really needs to understand how the wrapping really works is able to do so just from looking at the existing code.

Reply:
I think finding something in the documentation is probably easier than finding it in the code, especially for non-experienced Trixi.jl user. And we almost already explain what the precise structure of the wrapped array looks like. So there is not much to add, I would say.

Reply:
It sounds reasonable. Thank you!
I added following description:
#   This is why the `u_ode` vector is wrapped by the `wrap_array` function using `unsafe_wrap`
#   to form a multidimensional array `u`. In this array, the first dimension corresponds to
#   variables, followed by N dimensions corresponding to nodes for each of N space dimensions.
#   The last dimension corresponds to the elements. 
#   Consequently, navigation within this multidimensional array becomes noticeably easier.

Hopefully now it is better)

Reply:
Thanks a lot!